guides/deployment/releases.md

```diff
@@ -189,26 +189,41 @@ ENV MIX_ENV=prod
 
 # install mix dependencies
 COPY mix.exs mix.lock ./
-COPY config config
-RUN mix do deps.get --only $MIX_ENV, deps.compile
+RUN mix deps.get --only $MIX_ENV
+RUN mkdir config
+# if our compile-time config changes, we may need to
+# recompile dependencies
+COPY config/config.exs config/prod.exs config/
+RUN mix deps.compile
 
 # build assets
 COPY assets/package.json assets/package-lock.json ./assets/
+# install all npm dependencies from scratch
 RUN npm --prefix ./assets ci --progress=false --no-audit --loglevel=error
```

Comment:
I don't want to over-complicate an example, but oftentimes the node version needs to be specified for consistent builds, and the easiest way I can control this is by building the frontend assets in a separate container and copying the artifacts over.

For example:

```dockerfile
## BACKEND
FROM hexpm/elixir:1.11.2-erlang-23.1.4-alpine-3.12.1 AS builder
 
# ...

## FRONTEND

FROM node:14.14.0-alpine AS frontend
# PurgeCSS needs to see the Elixir stuff
COPY lib/my_app_web ./lib/my_app_web

# The package.json may include references to JavaScript inside Elixir dependencies. Copy them over:
COPY --from=builder /app/deps/phoenix ./deps/phoenix
COPY --from=builder /app/deps/phoenix_html ./deps/phoenix_html
COPY --from=builder /app/deps/phoenix_live_view ./deps/phoenix_live_view

COPY assets/package.json assets/package-lock.json ./assets/
RUN npm --prefix assets ci --progress=false --no-audit --loglevel=error
COPY assets ./assets
RUN npm --prefix assets run deploy

## APP

FROM builder AS app
COPY --from=frontend /app/priv/static ./priv/static
RUN mix phx.digest

COPY priv/gettext ./priv/gettext
COPY priv/repo ./priv/repo
```

Unfortunately doing it this way means I have to copy each folder under `/priv` separately, unless someone knows a more straightforward way to copy-merge them from the multi-stage builds.

WDYT of including a comment to mention frontend assets being a multi-stage build step if needing to control the nodejs version?

Reply:
@alfredbaudisch would you mind making a followup PR after this one is merged? You can tag me on it and I'll try to chime in.

Reply:
@dbernheisel would you mind making a followup PR after this one is merged? You can tag me on it and I'll try to chime in.